extensions/hotreload/hotreload.go

```diff
@@ -0,0 +1,104 @@
+package hotreload
+
+import (
+    "fmt"
+    "html/template"
+    "log"
+    "sync"
+
+    _ "embed"
+
+    . "github.com/emad-elsaid/xlog"
+    "github.com/gorilla/websocket"
+)
+
+var (
+    upgrader     = websocket.Upgrader{ReadBufferSize: 1024, WriteBufferSize: 1024}
+    clients      = make(map[*websocket.Conn]bool)
+    clientsMutex sync.Mutex
+)
+
+func init() {
+    if !READONLY {
```

Comment:
`READONLY` will be always false, as when `init()` is called the flags are not parsed yet. so I think you'll need to check `READONLY` variable inside each `handleWebSocket` and `clientWidget` to decide if they should return early.